diff -up chromium-80.0.3987.122/media/gpu/vaapi/BUILD.gn.i686permissive chromium-80.0.3987.122/media/gpu/vaapi/BUILD.gn
--- chromium-80.0.3987.122/media/gpu/vaapi/BUILD.gn.i686permissive	2020-02-27 11:58:10.534875341 -0500
+++ chromium-80.0.3987.122/media/gpu/vaapi/BUILD.gn	2020-02-27 11:58:17.544736535 -0500
@@ -13,6 +13,10 @@ import("//ui/ozone/ozone.gni")
 assert(is_linux)
 assert(use_vaapi)
 
+config("vaapi_permissive") {
+  cflags = [ "-fpermissive" ]
+}
+
 generate_stubs("libva_stubs") {
   extra_header = "va_stub_header.fragment"
   sigs = [ "va.sigs" ]
@@ -120,6 +124,8 @@ source_set("vaapi") {
     ]
   }
 
+  configs += [ ":vaapi_permissive" ]
+
   if (use_x11) {
     configs += [ "//build/config/linux:x11" ]
     deps += [ "//ui/gfx/x" ]
