diff -up chromium-83.0.4103.61/chrome/browser/search/background/ntp_backgrounds.h.missing-cstddef chromium-83.0.4103.61/chrome/browser/search/background/ntp_backgrounds.h
--- chromium-83.0.4103.61/chrome/browser/search/background/ntp_backgrounds.h.missing-cstddef	2020-05-18 14:39:11.000000000 -0400
+++ chromium-83.0.4103.61/chrome/browser/search/background/ntp_backgrounds.h	2020-05-29 15:37:52.370693881 -0400
@@ -6,6 +6,7 @@
 #define CHROME_BROWSER_SEARCH_BACKGROUND_NTP_BACKGROUNDS_H_
 
 #include <array>
+#include <cstddef>
 
 class GURL;
 
diff -up chromium-83.0.4103.61/media/cdm/supported_cdm_versions.h.missing-cstddef chromium-83.0.4103.61/media/cdm/supported_cdm_versions.h
diff -up chromium-83.0.4103.61/third_party/angle/include/platform/Platform.h.missing-cstddef chromium-83.0.4103.61/third_party/angle/include/platform/Platform.h
--- chromium-83.0.4103.61/third_party/angle/include/platform/Platform.h.missing-cstddef	2020-05-29 15:37:52.374693782 -0400
+++ chromium-83.0.4103.61/third_party/angle/include/platform/Platform.h	2020-05-29 15:39:47.343874233 -0400
@@ -12,6 +12,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <array>
+#include <cstddef>
 
 #define EGL_PLATFORM_ANGLE_PLATFORM_METHODS_ANGLEX 0x3482
 
diff -up chromium-83.0.4103.61/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h.missing-cstddef chromium-83.0.4103.61/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h
--- chromium-83.0.4103.61/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h.missing-cstddef	2020-05-18 14:42:43.000000000 -0400
+++ chromium-83.0.4103.61/third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h	2020-05-29 15:37:52.376693733 -0400
@@ -12,6 +12,7 @@
 #define MODULES_AUDIO_PROCESSING_AEC3_CLOCKDRIFT_DETECTOR_H_
 
 #include <array>
+#include <cstddef>
 
 namespace webrtc {
 
