diff -up chromium-84.0.4147.89/services/service_manager/sandbox/linux/bpf_cros_amd_gpu_policy_linux.cc.epel7-kcmp chromium-84.0.4147.89/services/service_manager/sandbox/linux/bpf_cros_amd_gpu_policy_linux.cc
--- chromium-84.0.4147.89/services/service_manager/sandbox/linux/bpf_cros_amd_gpu_policy_linux.cc.epel7-kcmp	2020-07-15 10:03:33.509319562 -0400
+++ chromium-84.0.4147.89/services/service_manager/sandbox/linux/bpf_cros_amd_gpu_policy_linux.cc	2020-07-15 10:12:55.761658501 -0400
@@ -6,7 +6,9 @@
 
 #include <errno.h>
 #include <fcntl.h>
-#include <linux/kcmp.h>
+#ifndef KCMP_FILE
+#define KCMP_FILE 0
+#endif
 #include <sys/socket.h>
 
 // Some arch's (arm64 for instance) unistd.h don't pull in symbols used here
