diff -up chromium-84.0.4147.89/base/check_op.h.el-constexpr chromium-84.0.4147.89/base/check_op.h
--- chromium-84.0.4147.89/base/check_op.h.el-constexpr	2020-07-18 09:38:18.700433097 -0400
+++ chromium-84.0.4147.89/base/check_op.h	2020-07-18 09:59:00.620767263 -0400
@@ -33,7 +33,7 @@ namespace logging {
 
 // Functions for turning check operand values into strings.
 // Caller takes ownership of the returned string.
-BASE_EXPORT char* CheckOpValueStr(int v);
+BASE_EXPORT constexpr char* CheckOpValueStr(int v);
 BASE_EXPORT char* CheckOpValueStr(unsigned v);
 BASE_EXPORT char* CheckOpValueStr(long v);
 BASE_EXPORT char* CheckOpValueStr(unsigned long v);
@@ -124,7 +124,7 @@ class CheckOpResult {
   // A non-success result. expr_str is something like "foo != bar". v1_str and
   // v2_str are the stringified run-time values of foo and bar. Takes ownership
   // of v1_str and v2_str.
-  BASE_EXPORT CheckOpResult(const char* expr_str, char* v1_str, char* v2_str);
+  BASE_EXPORT constexpr CheckOpResult(const char* expr_str, char* v1_str, char* v2_str);
 
   // Returns true if the check succeeded.
   constexpr explicit operator bool() const { return !message_; }
