diff -up chromium-84.0.4147.89/base/check_op.cc.el-constexpr chromium-84.0.4147.89/base/check_op.cc
diff -up chromium-84.0.4147.89/base/check_op.h.el-constexpr chromium-84.0.4147.89/base/check_op.h
--- chromium-84.0.4147.89/base/check_op.h.el-constexpr	2020-07-18 09:38:18.700433097 -0400
+++ chromium-84.0.4147.89/base/check_op.h	2020-07-18 10:51:04.086854722 -0400
@@ -171,7 +171,7 @@ class CheckOpResult {
     return ::logging::CheckOpResult(expr_str, CheckOpValueStr(v1),             \
                                     CheckOpValueStr(v2));                      \
   }                                                                            \
-  constexpr ::logging::CheckOpResult Check##name##Impl(int v1, int v2,         \
+  ::logging::CheckOpResult Check##name##Impl(int v1, int v2,                   \
                                                        const char* expr_str) { \
     if (ANALYZER_ASSUME_TRUE(v1 op v2))                                        \
       return ::logging::CheckOpResult();                                       \
